include/eve/module/algo/algo/transform_copy_if.hpp

```diff
@@ -0,0 +1,169 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/module/algo/algo/common_forceinline_lambdas.hpp>
+#include <eve/module/algo/algo/traits.hpp>
+#include <eve/module/algo/algo/two_stage_iteration.hpp>
+#include <eve/module/algo/algo/views/convert.hpp>
+#include <eve/module/core.hpp>
+
+#include <utility>
+#include <type_traits>
+
+namespace eve::algo
+{
+
+  namespace detail
+  {
+    template<typename TraitsSupport> struct transform_copy_if_ : TraitsSupport
+    {
+      template<typename OutIt, typename Func> struct delegate
+      {
+        OutIt out_first;
+        OutIt out_last;
+        Func func;
+
+        static constexpr auto compress_copy()
+        {
+          auto density = density_for_compress_copy<typename TraitsSupport::traits_type>();
+          return eve::compress_copy[eve::unsafe][density];
+        }
+
+        EVE_FORCEINLINE
+        bool tail(auto f, auto ignore)
+        {
+          auto out_ignore   = eve::keep_first(std::min(out_last - out_first, eve::iterator_cardinal_v<OutIt>));
+          auto loaded       = eve::load(f);
+          auto [vals, mask] = func(loaded);
+          out_first = eve::compress_store[eve::unsafe][ignore][out_ignore](vals, mask, out_first);
+          return out_first == out_last;
+        }
+
+        EVE_FORCEINLINE
+        std::ptrdiff_t left_for_stage1() const
+        {
+          return out_last - out_first;
+        }
+
+        EVE_FORCEINLINE
+        bool step_1(auto f)
+        {
+          auto loaded       = eve::load(f);
+          auto [vals, mask] = func(loaded);
+          out_first = eve::compress_store[eve::unsafe](vals, mask, out_first);
+          return false;
+        }
+
+        EVE_FORCEINLINE
+        bool step_2(auto f)
+        {
+          // ol - of < cardinal
+          auto out_ignore   = eve::keep_first(out_last - out_first);
+          auto loaded       = eve::load(f);
+          auto [vals, mask] = func(loaded);
+          out_first = eve::compress_store[eve::unsafe][eve::ignore_none][out_ignore](vals, mask, out_first);
+          return out_first == out_last;
+        }
+      };
+
+      template<relaxed_range In, relaxed_range Out, typename Func>
+      EVE_FORCEINLINE
+      auto operator()(In&& in, Out&& out, Func func) const -> unaligned_iterator_t<Out>
+      {
+        if( in.begin() == in.end() || out.begin() == out.end() )
+          return out.begin();
+
+        using out_type = eve::element_type_t<std::remove_reference_t<decltype(get<0>(func(std::declval<wide<value_type_t<In>>>())))>>;
```

Comment:
No, this is not right. This code means you miss a test for precise wide size being accepted.

You cannot call:

```
func(wide<int>{})
```

If the user passed in:

```
func(wide<int, eve::fixed<2>>)
```

That's just how we do this.

Solution:

You preprocess without convert and then convert later.
See transform
https://github.com/jfalcou/eve/blob/8969c8242499d6a91274931bfa16797254507464/include/eve/module/algo/algo/transform.hpp#L38